mmv1/third_party/terraform/tests/resource_sql_user_test.go

```diff
@@ -45,6 +45,45 @@ func TestAccSqlUser_mysql(t *testing.T) {
 	})
 }
 
+func TestAccSqlUser_mysqlDisabled(t *testing.T) {
+	// Multiple fine-grained resources
+	skipIfVcr(t)
+	t.Parallel()
+
+	instance := fmt.Sprintf("i-%d", randInt(t))
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccSqlUserDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testGoogleSqlUser_mysql(instance, "password", true),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleSqlUserExists(t, "google_sql_user.user1"),
+					testAccCheckGoogleSqlUserExists(t, "google_sql_user.user2"),
+					resource.TestCheckResourceAttr("google_sql_user.user1", "sql_server_user_details.disabled", "true"),
+				),
+			},
+			{
+				// Update password
+				Config: testGoogleSqlUser_mysql(instance, "password", false),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleSqlUserExists(t, "google_sql_user.user1"),
+					testAccCheckGoogleSqlUserExists(t, "google_sql_user.user2"),
+					resource.TestCheckResourceAttr("google_sql_user.user1", "sql_server_user_details.disabled", "false"),
+				),
+			},
+			{
+				ResourceName:            "google_sql_user.user2",
+				ImportStateId:           fmt.Sprintf("%s/%s/gmail.com/admin", getTestProjectFromEnv(), instance),
```

Comment:
I'll be completely honest, i have been unable to set up an environment to actually build or run any of this stuff - even via cloudshell does not appear to work.  i'm just trying to lift someone else's code into this project so I can get full access to the google API via terraform.

You requested a test, so i copied an existing test and added the disabled state change validation as well. i've committed your suggested changes, but am not sure what else you are looking for here.  Please feel free to make any other changes you desire, so we can move forward with using terraform to create Workload Identity SQL users, since it is effectively broken without this functionality.